test/model/settings_test.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/model/binding.dart';
+import 'package:zulip/model/settings.dart';
+
+import '../example_data.dart' as eg;
+import 'store_checks.dart';
+import 'store_test.dart';
+
+void main() {
+  final httpLink = Uri.parse('http://chat.zulip.org');
+  final nonHttpLink = Uri.parse('mailto:chat@zulip.org');
+
+  group('getUrlLaunchMode', () {
+    testAndroidIos('globalSettings.browserPreference is null; use our per-platform defaults for HTTP links', () {
+      final globalStore = eg.globalStore(globalSettings: eg.globalSettings(
+        browserPreference: null));
```

Comment:
nit: this bit of setup should arrive in the "follow setting" commit, rather than the preceding commit where the setting isn't yet involved